### PR TITLE
doc: security: Add information about backports

### PR DESCRIPTION
**Pull request by ceolin**
Clarify about how to proceed with backports for vulnerabilities.

### CODE REVIEW

**Review by d3zd3z — DISMISSED** [review]

File: doc/security/reporting.rst [hunk 1]
```diff
@@ -180,7 +182,12 @@ following releases:
 The developer of the fix shall be responsible for any necessary
 backports, and apply them to any of the above listed release branches,
 unless the fix does not apply (the vulnerability was introduced after
-this release was made).
+this release was made). All recommendations for
+:ref:`vulnerability fixes <vulnerability_fix_recommendation>` apply
+for backport pull request (and associated issue). Additionally is
```
Comment (nashif) [inline doc/security/reporting.rst:187]: pull requests, associated issues...

Additionally, it is recommended

Comment (ceolin) [reply doc/security/reporting.rst]: done

**Comment by ceolin** [comment]
@nashif can you please revisit it.

File: doc/security/reporting.rst [hunk 2]
```diff
@@ -180,7 +182,12 @@ following releases:
 The developer of the fix shall be responsible for any necessary
 backports, and apply them to any of the above listed release branches,
 unless the fix does not apply (the vulnerability was introduced after
-this release was made).
+this release was made). All recommendations for
+:ref:`vulnerability fixes <vulnerability_fix_recommendation>` apply
+for backport pull requests (and associated issues). Additionally, it is
+recommended that the developer privately inform the responsible
```
Comment (kartben) [inline doc/security/reporting.rst:188]:
```suggestion
recommended that the developer privately informs the responsible
```

File: doc/security/reporting.rst [hunk 3]
```diff
@@ -107,6 +107,8 @@ within the Zephyr project to fix the issues, and 60 days for external
 parties building products using Zephyr to be able to apply and
 distribute these fixes.
 
+.. _vulnerability_fix_recommendation:
```
Comment (kartben) [inline doc/security/reporting.rst:110]:
```suggestion
.. _vulnerability_fix_recommendations:
```

File: doc/security/reporting.rst [hunk 4]
```diff
@@ -180,7 +182,12 @@ following releases:
 The developer of the fix shall be responsible for any necessary
 backports, and apply them to any of the above listed release branches,
 unless the fix does not apply (the vulnerability was introduced after
-this release was made).
+this release was made). All recommendations for
+:ref:`vulnerability fixes <vulnerability_fix_recommendation>` apply
```
Comment (kartben) [inline doc/security/reporting.rst:186]:
```suggestion
:ref:`vulnerability fixes <vulnerability_fix_recommendations>` apply
```

**Review by kartben — CHANGES_REQUESTED** [review]
minor tweaks but looking good to me - thx!

**Comment by ceolin** [comment]
@kartben Thanks for reviewing it ! Fixed

**Review by kartben — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**